nadeo_event_api/src/nadeo_event_api/api/structure/settings/script_settings.py

```diff
@@ -2,17 +2,42 @@
 
 from ...structure.enums import RespawnBehavior
 
+# Nadeo api documentation: https://wiki.trackmania.io/en/dedicated-server/Usage/OfficialGameModesSettings
 
-class ScriptSettings(ABC):
+# 
+class BaseScriptSettings():
     def __init__(
-        self,
+        self,         
         chat_time: int = 10,
         force_laps_number: int = -1,
         respawn_behavior: RespawnBehavior = RespawnBehavior.DEFAULT,
         warmup_duration: int = 0,
         warmup_number: int = 0,
         warmup_timeout: int = -1,
         pick_ban_enable: bool = False,
+                 ):
+        
```

Comment:
A comment about how these are base settings that are shared among all gamemodes would be nice, you can look  at the format of the rest of the ScriptSetting classes as a guide